s3stream/src/main/java/com/automq/stream/s3/wal/impl/block/BlockWALService.java

```diff
@@ -115,7 +116,11 @@ public class BlockWALService implements WriteAheadLog {
     public static final int WAL_HEADER_CAPACITY = WALUtil.BLOCK_SIZE;
     public static final int WAL_HEADER_TOTAL_CAPACITY = WAL_HEADER_CAPACITY * WAL_HEADER_COUNT;
     private static final Logger LOGGER = LoggerFactory.getLogger(BlockWALService.class);
-    private final AtomicBoolean started = new AtomicBoolean(false);
+    private static final int WAL_STATE_INIT = 1;
+    private static final int WAL_STATE_STARTED = 2;
+    private static final int WAL_STATE_SHUTING_DOWN = 3;
+    private static final int WAL_STATE_SHUTDOWN = 4;
```

Comment:
If we use `AtomicReference` to maintain the state of `WAL` (as I mentioned below), then these `int` constants can also be replaced with a separate enum class.